testexec/testexec.go

```diff
@@ -50,6 +51,18 @@ type FilterFn func(line string) (replacement string)
 // The default behavior, if a Tester object doesn't get an AssertFn, is to fall back to a very basic call to `t.Errorf`.
 type AssertFn func(t *testing.T, actual, expect string)
 
+var (
+	// SkipRecursion can be returned by a RecursionFn for paths that should show up in the test as skipped
+	SkipRecursion = errors.New("skip this directory")
+	// IgnoreRecursion can be returned by a RecursionFn for paths that should not run a new test at all
+	IgnoreRecursion = errors.New("ignore this directory")
+)
+
+// RecursionFn can be used in Tester to specify a recursion pattern
+// The default recursion pattern will recurse whenever it finds "then-"
+// Returning a non-nil error will cause the test for the directory to fail unless the error is a SkipRecursion or IgnoreRecursion error
+type RecursionFn func(*testing.T, testmark.DirEnt) error
```

Comment:
I do want to fiddle with how the `/net/` flag works in warpforge and this would solve that particular use case. There are alternative ways to do that there though.

Reply:
Given the pivot to focusing this PR on strict mode, I think this type (and also the magic error vars above) can go away now?

Reply:
yeah I can fold the recursion function back into the loop if desired.